src/server/handlers/health_check.rs

```diff
@@ -0,0 +1,20 @@
+use actix_web::{web, HttpResponse};
+use serde::Serialize;
+
+use crate::state::State;
+
+pub async fn health_check(state: web::Data<State>) -> HttpResponse {
+    match sqlx::query("SELECT 1")
+        .fetch_one(state.db_pool.as_ref())
+        .await
+    {
+        Ok(_) => HttpResponse::Ok().json(ServiceStatus { db_status: true }),
+        Err(_) => HttpResponse::InternalServerError().json(ServiceStatus { db_status: false }),
+    }
+}
+
+/// Service Health Check
+#[derive(Serialize)]
+struct ServiceStatus {
+    db_status: bool,
+}
```

Comment:
Lets avoid use before implementations in our files, that makes reading the code more natural without the need of scrolling to find definitions.

I suggest moving this struct definition before its usage in `health_check` function.

This struct is only being used as a response here, so why not appending `Response` word at the end? `HealthCheckResponse` would be more semantic, WDYT?